src/components/NewLocationPage/CountyMap/CountyMap.tsx

```diff
@@ -1,17 +1,92 @@
-import React from 'react';
+import React, { useState } from 'react';
 import RegionMap from 'components/RegionMap';
 import FixedAspectRatio from 'components/FixedAspectRatio/FixedAspectRatio';
-import { MapContainer } from './CountyMap.style';
+import {
+  MapContainer,
+  ThermometerContainer,
+  ToggleWrapper,
+  Row,
+  VaccinationsThermLabel,
+} from './CountyMap.style';
 import { Region } from 'common/regions';
+import {
+  ButtonGroup,
+  Button,
+} from 'components/SharedComponents/SharedComponents.style';
+import { trackEvent, EventAction, EventCategory } from 'components/Analytics';
+import {
+  RiskLevelThermometer,
+  VaccinationsThermometer,
+} from 'components/HorizontalThermometer';
+import { getAlertColor } from 'components/RegionMap/RegionMap.style';
+import { LocationSummary } from 'common/location_summaries';
+import { vaccineColorFromLocationSummary } from 'common/colors';
+
+enum MapType {
+  VACCINATIONS = '% Vaccinated',
+  RISK_LEVEL = 'Risk levels',
+}
+
+interface MapTypeInfo {
+  thermometer: React.ReactNode;
+  colorMap: (locationSummary: LocationSummary) => string;
+}
+
+const MAP_TYPE_INFO: { [key in MapType]: MapTypeInfo } = {
+  [MapType.VACCINATIONS]: {
+    thermometer: (
+      <Row>
+        <VaccinationsThermLabel>
+          Pop. with
+          <br />
+          <strong>1+ dose</strong>
+        </VaccinationsThermLabel>
+        <VaccinationsThermometer />
+      </Row>
+    ),
+    colorMap: (locationSummary: LocationSummary) =>
+      vaccineColorFromLocationSummary(locationSummary),
+  },
+  [MapType.RISK_LEVEL]: {
+    thermometer: <RiskLevelThermometer />,
+    colorMap: (locationSummary: LocationSummary) =>
+      getAlertColor(locationSummary),
+  },
+};
 
 /* The default aspect-ratio for the state and US maps is 800x600 */
 
 const CountyMap: React.FC<{ region: Region }> = React.memo(({ region }) => {
+  const [mapType, setMapType] = useState<MapType>(MapType.RISK_LEVEL);
+
+  const onClickToggle = (
+    event: React.MouseEvent<HTMLElement>,
+    newView: MapType,
+  ) => {
+    if (newView) {
```

Comment:
nit: Looking at this code, `newView` is a `MapType` which will always be '% Vaccinated' or 'Risk levels' and so this `if` should always be true and therefore I might be tempted to remove it.

But my guess is it _is_ actually necessary (or could be anyway) because material-ui [docs](https://material-ui.com/api/toggle-button-group/) say that onChange can be called with `null`... so the type should probably be `newView: MapType|null` and then it's clearer why the `if` statement is needed.

As an aside, the reason you were able to type it as `MapType` even though that's not quite right is because the material-ui code is typed as:
```
  onChange?: (event: React.MouseEvent<HTMLElement>, value: any) => void;
```

Since they used `any`, TypeScript isn't able to help us type it correctly.

Reply:
ah yes, it breaks when removing the if--the possible `null` is probably why